[13.0] Конечные изменения

em230418 opened the pull request.

em230418 committed:
:zap: saas: added ability to redefine mandatory modules

em230418 committed:
:tada: :one::two: saas_expiration

em230418 committed:
:tada: :one::two: saas_limit_max_users

em230418 committed:
:zap: saas: added ability to get refresh info from build databases

em230418 committed:
:tada: :one::two: saas_access_apps

em230418 committed:
:ambulance: saas: random_ready_operator does not throw exceptions if no operators are ready

em230418 committed:
:tada: :one::two: saas_portal

em230418 committed:
:ambulance: saas: no extra right required to read build's url

em230418 committed:
:heart_eyes: saas: removed db_id variable from database name template

em230418 committed:
:zap: saas: added execute_kw and xmlid_lookup on saas.db

em230418 committed:
:arrow_up: :one::three: saas_access_apps

em230418 committed:
:arrow_up: :one::three: saas_expiration

em230418 committed:
:arrow_up: :one::three: saas_limit_max_users

em230418 committed:
:arrow_up: :one::three: saas_portal

em230418 committed:
:tada: :one::three: saas_product

em230418 committed:
:ambulance: saas: do not try to update global_url on draft builds

em230418 committed:
:heart_eyes: saas: added option to deploy database using draft database records

em230418 committed:
:tada: :one::three: saas_build_admin

em230418 committed:
:ambulance: saas: fix 'Connect to the Build' button when build admin's login differs from 'admin'

em230418 committed:
:tada: :one::three: saas_database_limit

em230418 committed:
:tada: :one::three: saas_operator_remote

em230418 committed:
:heart_eyes: saas: added "Operators" menu

em230418 committed:
:heart_eyes: saas.operator.name field is now mandatory

em230418 committed:
:heart_eyes: saas: reorganized saas_operator code for more comfortable derives

em230418 committed:
:shield: update oca_dependencies.txt

em230418 committed:
:book: saas_build_admin: add section about configurating SMTP server

em230418 committed:
:ambulance: saas: turn off tests when installing modules and creating database

em230418 committed:
:book: saas_operator_remote: add missing README.rst

em230418 committed:
:heart_eyes: saas: add button_box to builds

em230418 committed:
:heart_eyes: saas: added "Install missing mandatory modules" button in build and template operator views on developer mode

em230418 committed:
:book: saas_portal: added known bug about number of queue jobs

em230418 committed:
:tada: :one::three: saas_apps

Co-authored-by: Vildan Safin <safin@it-projects.info>
Co-authored-by: Eugene Molotov <molotov@it-projects.info>

em230418 committed:
:tada: :one::three: saas_contract

em230418 committed:
:tada: :one::three: saas_apps_signup

yelizariev commented on saas_product/doc/changelog.rst:
```diff
@@ -0,0 +1,4 @@
+`1.0.0`
+-------
+
+- **Init version**
```
Этот бот еще живой, оказывается)

em230418 committed:
:shield: saas_apps: disable js-empty-coverage

em230418 committed:
:shield: added missing server-auth repository for oca_depenencies

em230418 committed:
:shield: added missing contract repository for oca_dependencies

em230418 committed:
:ambulance: saas: flush job after writing it

yelizariev commented on saas/models/queue_job.py:
```diff
@@ -0,0 +1,10 @@
+
+    def write(self, vals):
+        res = super(QueueJob, self).write(vals)
+        self.flush()
```
Зачем? И нужно ли это обновить в oca/queue?

em230418 replied:
В прошлом году я создавал https://github.com/OCA/queue/pull/215, в котором описал проблему с которой я сталкиваюсь. Если вкратце, состояние job по-факту завершает работу, в логах записывается что job завершен, а в базе он висит как "выполняется".

Изначально было там другой commit. Суть того коммита был вызов `db_record.flush()` после вызова `db_record.write()` в той строке, которая сейчас выглядит так: https://github.com/OCA/queue/blob/87346515fd0cd2080f4793d19c85be05e25affe0/queue_job/job.py#L538-L540

Сам PR я создал в середине марта. В октябре сопровождающий после моего запроса решил мимолетом взглянуть на мой PR. Далее сопровождающий сделал предложение поменять commit, ибо не хочет вызывать flush в том месте и описал почему.

В итоге я сделал другой коммит, который на тот момент решал задачу. Я проверял, что новый коммит действительно решал мою задачу. И в итоге это изменение сопровождающий принял. Это была середина октября.

Далее в конце того года делаю рефакторинг, обнаруживаю, что с принятым коммитом задача остается нерешенной. И я не представляю, почему мне казалось в середине октября, что она решена. Может быть применил новый коммит к старой версии queue_job, когда еще там декораторы `@job` были обязательны, и получилось что все работает. Но серьезно тогда копать не было времени.

Сейчас появились другие клиенты, и опять сталкиваюсь с задачей, где job выполняется, в логах пишет что закончил, а по факту не записывает.

Надо ли в OCA/queue_job создавать PR? Надо. Но у меня нет времени на много что. Например:
- выяснение причины, почему принятый commit не решил мою задачу (по сути отладка)
- на оформление PR
- обсуждение с сопровождающим

Но в тоже время я знаю как решить задачу быстро. flush и все нафиг.

yelizariev replied:
Может ишью открыть хотя бы? Есть шаги воспроизведения?

em230418 committed:
:shield: travis: several fixes:

- Use special maintainer-quality-tools fork
- Enabled OCA/OCB
- Disabled manifest-name warning
- Move OPTIONS env variable for github to show correctly build configuration

em230418 committed:
:green_heart: saas: ignore odoo 13.0+ specific lint error

em230418 committed:
:green_heart: saas_apps_signup: ignore odoo 13.0+ specific lint error

em230418 committed:
:green_heart: saas_apps: remove duplicated view_mode field

em230418 committed:
:green_heart: saas_contract: allow to translate exception message

em230418 committed:
:green_heart: saas_apps_signup: add python-slugify to external-dependencies

em230418 committed:
:ambulance: saas_public: fix 'Wrong token' return, when public user is redirected to created build

em230418 committed:
:ambulance: saas_apps_signup: when build is made by contract, inappropriate template operator can be used

em230418 committed:
:ambulance: saas_apps_signup: when build is made by contract, incorrect exception message is given when no template operators are set in used template
